cmd/nexusUpload.go

```diff
@@ -0,0 +1,256 @@
+	if err == nil {
+		artifactID, err = evaluateMavenProperty(pomFile, "project.artifactId")
+	}
+	var artifactsVersion string
+	if err == nil {
+		artifactsVersion, err = evaluateMavenProperty(pomFile, "project.version")
+	}
+	if err == nil {
+		err = nexusClient.SetArtifactsVersion(artifactsVersion)
+	}
+	if err == nil {
+		artifact := nexus.ArtifactDescription{
+			File:       pomFile,
+			Type:       "pom",
+			Classifier: "",
+			ID:         artifactID,
+		}
+		err = nexusClient.AddArtifact(artifact)
+	}
+	if err == nil {
+		err = addTargetArtifact(pomFile, targetFolder, artifactID, nexusClient)
+	}
+	if err == nil {
+		err = addAdditionalClassifierArtifacts(additionalClassifiers, targetFolder, artifactID, nexusClient)
+	}
+	if err == nil {
+		err = nexusClient.UploadArtifacts()
+	}
+	return err
+}
+
+func addTargetArtifact(pomFile, targetFolder, artifactID string, nexusClient *nexus.Upload) error {
+	packaging, err := evaluateMavenProperty(pomFile, "project.packaging")
+	if err != nil {
+		return err
+	}
+	if packaging == "pom" {
+		// Only pom.xml itself is the artifact
+		return nil
+	}
+	if packaging == "" {
+		packaging = "jar"
+	}
+	finalName, err := evaluateMavenProperty(pomFile, "project.build.finalName")
+	if err != nil || finalName == "" {
+		// NOTE: The error should be ignored, and the finalName built as Maven would from artifactId and so on.
+		// But it seems this expression always resolves, even if finalName is nowhere declared in the pom.xml
+		return err
+	}
+	filePath := composeFilePath(targetFolder, finalName, packaging)
+	artifact := nexus.ArtifactDescription{
+		File:       filePath,
+		Type:       packaging,
+		Classifier: "",
+		ID:         artifactID,
+	}
+	return nexusClient.AddArtifact(artifact)
+}
+
+func addAdditionalClassifierArtifacts(additionalClassifiers, targetFolder, artifactID string, nexusClient *nexus.Upload) error {
+	if additionalClassifiers == "" {
+		return nil
+	}
+	classifiers, err := getClassifiers(additionalClassifiers)
+	if err != nil {
+		return err
+	}
+	for _, classifier := range classifiers {
+		if classifier.Classifier == "" || classifier.FileType == "" {
+			return fmt.Errorf("invalid additional classifier description (classifier: '%s', type: '%s')",
+				classifier.Classifier, classifier.FileType)
+		}
+		filePath := composeFilePath(targetFolder, artifactID+"-"+classifier.Classifier, classifier.FileType)
+		artifact := nexus.ArtifactDescription{
+			File:       filePath,
+			Type:       classifier.FileType,
+			Classifier: classifier.Classifier,
+			ID:         artifactID,
+		}
+		err = nexusClient.AddArtifact(artifact)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func composeFilePath(folder, name, extension string) string {
+	path := name + "." + extension
+	if folder != "" {
+		path = folder + "/" + path
+	}
+	return path
+}
+
+func evaluateMavenProperty(pomFile, expression string) (string, error) {
+	execRunner := command.Command{}
+	execRunner.Stdout(ioutil.Discard)
```

Comment:
is it really intended to hide the maven output?

Reply:
Intended yes, since the output is the plain evaluation result of the expression. (The Defines further below include "-q".) But I might be missing something why this is still wrong.